keps/sig-cluster-lifecycle/kubeadm/20191229-kubeadm-move-out-of-kk.md

```diff
@@ -0,0 +1,594 @@
+as long as the CI build process continues to push the kubeadm
+and the rest of the k8s binaries to the GCS buckets.
+
+### Enhancements proposals
+
+The current plan is for kubeadm to continue following the k/enhancements
+and KEP process, given the project will be part of the k8s release.
+There will be no extra burden on the release team as they are already
+following PRs and issues in repositories other than k/k.
+
+There is also the option to decouple the enhancement process and have future
+kubeadm KEPs in the k/kubeadm repository, but needs further discussion
+and there are no apparent benefits.
+
+### Documentation
+
+#### Authored content
+
+There are no planned changes for how kubeadm will release its authored
+documentation. Kubeadm already has multiple documents in the k8s.io website
+and those are maintained with the help of sig-docs.
+
+#### Reference documentation
+
+The reference documentation update process needs changes.
+Currently tooling maintained by sig-docs does the following:
+1) Clones k/k
+2) Runs tools to generate cmd/kubeadm/app Cobra reference documentation
+3) Prettyfies the documentation with HTML/CSS
+
+The tooling needs to change to start generating the documentation
+from the k/kubeadm repository. SIG docs must be notified of this
+and collaborate with the kubeadm maintainers when the move is about to happen.
+
+There would not be a need for the tooling to continue building old
+reference documentation from k/k instead, as the k8s.io website reference
+documentation is only generated once before a release happens.
+
+The tools for building k/kubeadm reference docs in k/k must be removed.
+The initial proposal is to not support building reference documentation
+from the tools in the k/kubeadm repository, but might need some discussion.
+Currently the consumers of k/k reference documentation are unknown and the
+documentation is no longer tracked in commits. See:
+https://github.com/kubernetes/kubernetes/tree/master/docs
+
+
+#### Dependency updates
+
+With the kubeadm moves out of k/k, the project will be decoupled
+from the /vendor and /staging dependencies that it currently has in k/k.
+This poses the question of how often the kubeadm project will have to
+update its k8s.io/ dependencies in the go.mod file.
+
+Example of a go.mod file for kubeadm:
+https://github.com/neolit123/kubeadm/blob/66b94f324b63005589bab7132193e8031001ba57/kubeadm/go.mod
+
+The current proposal is to manually update these dependencies right
+after each release or during a release cycle if something is critical.
+The k/k issue tracker and Slack will be the source of signal if something
+has to be updated.
+
+##### Scenarios:
+
+###### A bug in k8s.io/[some-library] is found and fixed in /staging
+or /vendor of k/k
+
+The kubeadm maintainers need to watch for such changes or bug reports
+and update the kubeadm go.mod file. Running the change in CI will confirm
+that it is not regressing.
```

Comment:
Comparing the go.mod and go.sum files between k/k and k/kubeadm is easy, but mechanical. We can automate that.

Reply:
I would at least try to automate the monitoring for all those cases and have an alert fired somewhere so those events are not missed

Reply:
leaving this automation for KEP followups for the future after the move.